### PR TITLE
Added --instrumentation_template switch

### PR DESCRIPTION
**Pull request by mweissbacher**
Referencing https://github.com/google/closure-compiler/issues/1173#event-425562491

### CODE REVIEW

File: src/com/google/javascript/jscomp/CommandLineRunner.java [hunk 1]
```diff
@@ -1185,6 +1195,31 @@ protected CompilerOptions createOptions() {
 
     options.setConformanceConfigs(loadConformanceConfigs(flags.conformanceConfigs));
 
+    if ( ! flags.instrumentationFile.isEmpty()) {
+        String instrumentationPb;
+        Instrumentation.Builder builder = Instrumentation.newBuilder();
+        try {
```
Comment (dimvar) [inline src/com/google/javascript/jscomp/CommandLineRunner.java]: Can you convert this try into a try-with-resources?
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

File: src/com/google/javascript/jscomp/CompilerOptions.java [hunk 2]
```diff
@@ -920,6 +920,8 @@ public void setProtectHiddenSideEffects(boolean enable) {
    */
   public boolean instrumentForCoverage;
 
+  public String instrumentationTemplateFile;
```
Comment (dimvar) [inline src/com/google/javascript/jscomp/CompilerOptions.java]: Make this field package private.

**Comment by dimvar** [comment]
Looks good. I just had a couple of small comments.

File: src/com/google/javascript/jscomp/CommandLineRunner.java [hunk 3]
```diff
@@ -1185,6 +1195,29 @@ protected CompilerOptions createOptions() {
...
+                sb.append(System.lineSeparator());
+                line = br.readLine();
+            }
+            instrumentationPb = sb.toString();
+            TextFormat.merge(instrumentationPb, builder);
+
+            // Setting instrumentation template
+            options.instrumentationTemplate = builder.build();
+
+          } catch (IOException e) {
```
Comment (dimvar) [inline src/com/google/javascript/jscomp/CommandLineRunner.java]: The catch block should be indented a couple of spaces in, same column as the try.

File: src/com/google/javascript/jscomp/CommandLineRunner.java [hunk 4]
```diff
@@ -1185,6 +1195,29 @@ protected CompilerOptions createOptions() {
 
     options.setConformanceConfigs(loadConformanceConfigs(flags.conformanceConfigs));
 
+    if ( ! flags.instrumentationFile.isEmpty()) {
```
Comment (dimvar) [inline src/com/google/javascript/jscomp/CommandLineRunner.java]: can you remove the spaces before and after the !

**Comment by dimvar** [comment]
Hey Michael, looks like github doesn't send an email when you update the code. Can you post on this thread when you make the formatting changes I just requested, so I can merge the PR?

**Comment by mweissbacher** [comment]
Interesting, I was assuming it would do that. I fixed these two formatting issues, and a third one which I just noticed.

**Pull request merged**